File: app/src/main/java/com/dkarakaya/songify/MainActivity.java
package com.dkarakaya.songify;

import android.app.ActivityManager;
import android.app.NotificationManager;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    static TextView tvCurSongTitle;
    static TextView tvCurArtist;
    static LinearLayout lCurSongDetails;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvCurSongTitle =findViewById(R.id.tvCurSongTitle);
        tvCurArtist =findViewById(R.id.tvCurSongArtist);
        lCurSongDetails=findViewById(R.id.curSongDetails);
        lCurSongDetails.setVisibility(View.GONE);


        BottomNavigationView bottomNav = findViewById(R.id.bottom_navigation);
        bottomNav.setOnNavigationItemSelectedListener(navListener);

        //added this if statement to keep the selected fragment when rotating the device
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                    new LibraryFragment()).commit();
        }

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

    }

    private BottomNavigationView.OnNavigationItemSelectedListener navListener =
            new BottomNavigationView.OnNavigationItemSelectedListener() {
                @Override
                public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                    Fragment selectedFragment = null;

                    switch (item.getItemId()) {
                        case R.id.nav_lib:
                            selectedFragment = new LibraryFragment();
                            break;
                        case R.id.nav_search:
                            selectedFragment = new SearchFragment();
                            break;
                    }
                    getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container,
                            selectedFragment).commit();
                    return true;
                }
            };

    public static void setCurSongDetails(String curSongTitle, String curSongArtist){
        tvCurSongTitle.setText(curSongTitle);
        tvCurArtist.setText(curSongArtist);
        lCurSongDetails.setVisibility(View.VISIBLE);
    }
}

File: app/src/main/java/com/dkarakaya/songify/MainActivity.kt
package com.dkarakaya.songify

import android.os.Bundle
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import com.dkarakaya.songify.library.LibraryFragment
import com.dkarakaya.songify.search.SearchFragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : FragmentActivity(R.layout.activity_main) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        render()
        //added this if statement to keep the selected fragment when rotating the device
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().replace(R.id.container,
                    LibraryFragment()).commit()
        }
    }

    private fun render() {
        songTitle = cur_song_title
        songArtist = cur_song_artist
        val bottomNav = findViewById<BottomNavigationView>(R.id.bottom_navigation)
        bottomNav.setOnNavigationItemSelectedListener(navListener)
    }

    private val navListener: BottomNavigationView.OnNavigationItemSelectedListener =
            BottomNavigationView.OnNavigationItemSelectedListener { item ->
                lateinit var fragment: Fragment
                when (item.itemId) {
                    R.id.nav_lib -> fragment = LibraryFragment()
                    R.id.nav_search -> fragment = SearchFragment()
                }
                supportFragmentManager.beginTransaction().replace(R.id.container,
                        fragment).addToBackStack(null).commit()
                true
            }


    companion object {
        lateinit var songTitle: TextView
        lateinit var songArtist: TextView

        fun setPlayingSongDetails(title: String?, artist: String?) {
            songTitle.text = title
            songArtist.text = artist
        }
    }
}
